m/main_glider_data_processing_dt.m

```diff
@@ -589,143 +653,160 @@
-        end
-        if status
-          [success, message] = copyfile(output_local_file, output_public_file);
-          if success
-            disp(['Public output ' output_name ' succesfully copied: ' ...
-                  output_public_file '.']);
-          else
-            disp(['Error creating public copy of deployment product ' ...
-                  output_name ': ' output_public_file '.']);
-            disp(message);
-          end
-        else
-          disp(['Error creating public output directory ' ...
-                output_public_dir ':']);
-          disp(message);
-        end
-      end
+  %% Generate L2 matlab file, if needed and possible.
+  if ~isempty(fieldnames(data_gridded)) && ~isempty(matlab_l2_file)
+    disp('Generating Matlab L2 output...');
+    try
+      outputs.matlab_l2 = generateOutputMatlab( ...
+          matlab_l2_file, data_processed, meta_processed, ...
+          data_gridded, meta_gridded, deployment);
+      disp(['Output Matlab L2 generated: ' ...
+            outputs.matlab_l2 '.']);
+    catch exception
+      disp(['Error generating Matlab L2 output ' ...
+            matlab_l2_file ':']);
+      disp(getReport(exception, 'extended'));
     end
   end
+ 
+  %% Copy selected products to corresponding public location, if needed.
+% $$$   if ~isempty(fieldnames(outputs))
+% $$$     disp('Copying public outputs...');
+% $$$     output_name_list = fieldnames(outputs);
+% $$$     for output_name_idx = 1:numel(output_name_list)
+% $$$       output_name = output_name_list{output_name_idx};
+% $$$       if isfield(config.paths_public, output_name) ...
+% $$$            && ~isempty(config.paths_public.(output_name))
+% $$$         output_local_file = outputs.(output_name);
+% $$$         output_public_file = ...
+% $$$           strfstruct(config.paths_public.(output_name), deployment);
+% $$$         output_public_dir = fileparts(output_public_file);
+% $$$         [status, attrout] = fileattrib(output_public_dir);
+% $$$         if ~status
+% $$$           [status, message] = mkdir(output_public_dir);
+% $$$         elseif ~attrout.directory
+% $$$           status = false;
+% $$$           message = 'not a directory';
+% $$$         end
+% $$$         if status
+% $$$           [success, message] = copyfile(output_local_file, output_public_file);
+% $$$           if success
+% $$$             disp(['Public output ' output_name ' succesfully copied: ' ...
+% $$$                   output_public_file '.']);
+% $$$           else
+% $$$             disp(['Error creating public copy of deployment product ' ...
+% $$$                   output_name ': ' output_public_file '.']);
+% $$$             disp(message);
+% $$$           end
+% $$$         else
+% $$$           disp(['Error creating public output directory ' ...
+% $$$                 output_public_dir ':']);
+% $$$           disp(message);
+% $$$         end
+% $$$       end
+% $$$     end
+% $$$   end
+% $$$ 
```

Comment:
The creation of public copies of the selected products can be disabled in `configDTPathsLocal`.